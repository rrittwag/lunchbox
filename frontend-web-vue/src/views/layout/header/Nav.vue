<template>
  <nav class="flex items-center justify-center sm:h-16" aria-label="Haupt">
    <ul class="list-none">
      <slot></slot>
    </ul>
  </nav>
</template>
