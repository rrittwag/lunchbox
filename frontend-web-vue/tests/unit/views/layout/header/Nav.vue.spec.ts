import Nav from '@/views/layout/header/Nav.vue'
import { mount } from '@vue/test-utils'
import { describe, test, expect } from 'vitest'

describe('Nav', () => {
  test('renders snapshot', () => {
    const wrapper = mount(Nav, {
      slots: { default: '<li><a href="/">link1</a></li>  <li><a href="/">link2</a></li>' },
    })

    expect(wrapper.element).toMatchSnapshot()
  })

  test('is nav tag with aria-label (a11y)', () => {
    const wrapper = mount(Nav, {
      slots: { default: '<li><a href="/">link1</a></li>  <li><a href="/">link2</a></li>' },
    })

    const navTag = wrapper.get('nav')
    expect(navTag.attributes()['aria-label']).toBeDefined()
    wrapper.get('nav > ul') // throws error if not existing
  })

  test('renders slot NavLinks', () => {
    const wrapper = mount(Nav, {
      slots: { default: '<li><a href="/">link1</a></li>  <li><a href="/">link2</a></li>' },
    })

    const navLinks = wrapper.findAll('ul > li')
    expect(navLinks).toHaveLength(2)
    expect(navLinks[0].text()).toBe('link1')
    expect(navLinks[1].text()).toBe('link2')
  })
})
