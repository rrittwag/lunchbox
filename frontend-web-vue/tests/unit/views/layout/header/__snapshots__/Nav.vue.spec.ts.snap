// Vitest Snapshot v1

exports[`Nav > renders snapshot 1`] = `
<nav
  aria-label="Haupt"
  class="flex items-center justify-center sm:h-16"
>
  <ul
    class="list-none"
  >
    
    <li>
      <a
        href="/"
      >
        link1
      </a>
    </li>
     
    <li>
      <a
        href="/"
      >
        link2
      </a>
    </li>
    
  </ul>
</nav>
`;
